protoarray/src/test/java/tech/pegasys/teku/protoarray/ProtoArrayTest.java

```diff
@@ -464,19 +474,12 @@ private void assertStrictHead(final Bytes32 expectedBlockHash) {
   }
 
   private void addValidBlock(final long slot, final Bytes32 blockRoot, final Bytes32 parentRoot) {
-    addBlock(slot, blockRoot, parentRoot, false);
+    addOptimisticBlock(slot, blockRoot, parentRoot);
+    protoArray.markNodeValid(blockRoot);
```

Comment:
Changed to doing it this way to maintain the guarantee that all ancestors of a valid block are also fully validated. In production code this is ensured because blocks from phase0 and Altair are always valid when added and merge blocks are always optimistic when added. The test starts with the finalised node optimistic and many tests then add valid blocks on top.

`markNodeAsValid` will ensure all ancestors are also marked as valid but `onBlock` doesn't.